addons/mail/models/mail_thread.py

```diff
@@ -3167,15 +3167,17 @@ def _notify_thread_by_inbox(self, message, recipients_data, msg_vals=False, **kw
                 'res_partner_id': pid,
             } for pid in inbox_pids]
             self.env['mail.notification'].sudo().create(notif_create_values)
+            # sudo: res.partner to format personalized message
+            partners = self.env['res.partner'].sudo().browse(inbox_pids)
+            for partner in partners:
+                users = partner.user_ids
+                internal_users = users - users.filtered('share')
+                main_user = (internal_users or users or self.env['res.users'])[:1]
```

Comment:
Why would we have this kind of fallback when only internal users have access to inbox ?

Reply:
the internal part can be removed but there's a case where partner having more than one user_id, so we still need the users[:-1]